### PR TITLE
Subtraction of matrix in java

### PR DESCRIPTION
**Pull request by raksha009**
Issue no #1628 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Addition of new code
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Matrix_Subtraction.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raksha009** [comment]
@Vaibhavraj10 plz review my PR. Thanks

**Comment by raksha009** [comment]
Plz review my PR.
@asha15 @Vaibhavraj10 @raghav-dalmia @singh-shreya6

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        System.out.println("Of Second Matrix");
+        System.out.println(" Enter number of rows ");
+        int rows_second = s.nextInt();
+
+        System.out.println(" Enter number of columns ");
+        int columns_second = s.nextInt();
+        int[][] matrix2 = new int[rows_second][columns_second];
+
+        if (columns_first != columns_second || rows_first != rows_second) {
+            System.out.println(" Addition of matrix is not possible ");
```
Comment (asha15) [inline Matrix_Operations/Java/Matrix_Subtraction.java]: Addition?

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Java Program to calculation subtraction of 2 matrices (2D Array)
+import java.util.*;
+
+public class Matrix_Subtraction {
+
+    // Find subtraction of 2 Matrices and store it in matrix3 
+    public static void subtract_matrices(int[][] matrix1, int[][] matrix2) {
+        int index1, index2 = 0;
+        int[][] matrix3 = new int[matrix1.length][matrix1[0].length];
+        for ( index1 = 0; index1 < matrix1.length ; index1 ++) {
```
Comment (asha15) [inline Matrix_Operations/Java/Matrix_Subtraction.java]: remove the space after the bracket

Comment (asha15) [reply Matrix_Operations/Java/Matrix_Subtraction.java]: remove the space before the ;

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            for ( index2 = 0; index2 < matrix1[0].length; index2++ ) {
+                matrix3[index1][index2] = matrix1[index1][index2] - matrix2[index1][index2];
+            }
+        }
+
+        // Print the calculated matrix
+        for ( index1 = 0; index1 < matrix3.length ; index1 ++) {
```
Comment (asha15) [inline Matrix_Operations/Java/Matrix_Subtraction.java]: remove the space after the bracket
remove the space before the ;

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            System.out.println(" Enter elements of 2nd matrix ");
+            for (int i = 0 ; i < rows_first; i++) {
+                for (int j = 0 ; j< columns_first; j++) {
+                    matrix2[i][j] = s.nextInt();
+                }
+            }
+            subtract_matrices(matrix1, matrix2);
+        }
+    }
+}
```
Comment (asha15) [inline Matrix_Operations/Java/Matrix_Subtraction.java]: keep a line space after this

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+4
+ Enter elements of 2nd matrix 
+1
+3
+
+OUTPUT
+2 1 
+
+*/
+
```
Comment (asha15) [inline Matrix_Operations/Java/Matrix_Subtraction.java]: remove this line

**Comment by raksha009** [comment]
done the changes @asha15 maam

**Comment by asha15** [comment]
Have you send PR for all the matrix operation in java?

**Comment by raksha009** [comment]
not right now, i am working on few @asha15 maam

**Comment by raksha009** [comment]
Should I squash the commits in this PR!?

**Comment by asha15** [comment]
sure.

please send PR for all the operation in java and let me know. so it's easy for me to give scores for you

**Comment by raksha009** [comment]
squashed the commits @asha15 maam

**Comment by raksha009** [comment]
@asha15 maam, i made a few more PRs for this issue but i have to make more. so, since there are many operations, it would take time. Sorry for the delay. I'll try to cover as much as operations i could. Thanks

**Comment by raksha009** [comment]
@asha15 maam, i made a few more PRs for this issue but i have to make more. so, since there are many operations, it would take time. Sorry for the delay. I'll try to cover as much as operations i could. Thanks

**Comment by asha15** [comment]
NO problem

If you made all the operations in java, please let me know.

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        int[][] matrix2 = new int[rows_second][columns_second];
+
+        if (columns_first != columns_second || rows_first != rows_second) {
+            System.out.println(" Subtraction of matrix is not possible ");
+        }
+        else {
+
+            System.out.println(" Enter elements of 1st matrix ");
+            for (int i = 0 ; i < rows_first; i++) {
+                for (int j = 0 ; j< columns_first; j++) {
```
Comment (MastersAbh) [inline Matrix_Operations/Java/Matrix_Subtraction.java:51]: space before <

File: Matrix_Operations/Java/Matrix_Subtraction.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                    matrix1[i][j] = s.nextInt();
+                }
+            }
+
+            System.out.println(" Enter elements of 2nd matrix ");
+            for (int i = 0 ; i < rows_first; i++) {
+                for (int j = 0 ; j< columns_first; j++) {
```
Comment (MastersAbh) [inline Matrix_Operations/Java/Matrix_Subtraction.java:58]: space before <

**Comment by raksha009** [comment]
Done the changes @MastersAbh

**Comment by raksha009** [comment]
I have made all the PRs for all the operations in java. Plz review my PR @asha15 maam. Thanks

**Comment by raksha009** [comment]
I have made all the changes. plz merge my PR @singh-shreya6 @somya-kapoor @raghav-dalmia

**Pull request merged**